File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null) return null;
        if(head.val == val) {
            return removeElements(head.next, val);
        }
        ListNode curr = head;
        curr.next = removeElements(curr.next, val);
        return curr;
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if head == None: return head
        if head.val == val:
            while head and head.val == val:
                head = head.next
        curr = head
        prev = None
        while curr:
            if curr.val == val:
                while curr and curr.val == val:
                    curr = curr.next
                prev.next = curr
            prev = curr
            if curr: curr = curr.next
        return head